### PR TITLE
Add alnum (text pattern) multi field to host.name and observer.name (help wanted)

### PR DESCRIPTION
**Pull request by ypid-geberit**
The background is to make parts of the FQDN or hostname searchable. Example:

```
POST _analyze
{
  "analyzer": "pattern",
  "text": "host55-bmc.example.net"
}
```

Output:

```json
{
  "tokens" : [
    {
      "token" : "host55",
      "start_offset" : 0,
      "end_offset" : 6,
      "type" : "word",
      "position" : 0
    },
    {
      "token" : "bmc",
      "start_offset" : 7,
      "end_offset" : 10,
      "type" : "word",
      "position" : 1
    },
    {
      "token" : "example",
      "start_offset" : 11,
      "end_offset" : 18,
      "type" : "word",
      "position" : 2
    },
    {
      "token" : "net",
      "start_offset" : 19,
      "end_offset" : 22,
      "type" : "word",
      "position" : 3
    }
  ]
}
```

Ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-analyzers.html

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)?
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)?
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)?

### CODE REVIEW

File: scripts/generators/es_template.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def entry_for(field):
                 if mf_type == 'keyword':
                     ecs_helpers.dict_copy_existing_keys(mf, mf_entry, ['normalizer', 'ignore_above'])
```
Comment (vpiserchia) [inline scripts/generators/es_template.py:176]: why not supporting the ```analyzer``` key here too?

Comment (ypid-geberit) [reply scripts/generators/es_template.py]: Done that. I did not work much with different analyzers in index templates so I was not sure.

**Comment by kgeller** [comment]
@ypid-geberit I think this may be better suited for an RFC. It definitely seems like a straightforward addition, but I think it would be best to get others to weigh in. What do you think?

**Comment by ypid-geberit** [comment]
Thanks, sounds good. I have not gotten around to do it. If someone has the same need, feel free to continue my work. Otherwise, I cannot make promises but maybe in a few months I can consider it again.

**Comment by kgeller** [comment]
Going to close this out for now. Feel free to re-open in the future.

**Pull request closed without merging**